app/src/main/java/com/checkin/app/checkin/Home/NearbyRestaurantAdapter.kt

```diff
@@ -1,43 +1,155 @@
 package com.checkin.app.checkin.Home
 
+import android.app.PendingIntent.getActivity
+import android.content.Context
+import android.util.Log
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.fragment.app.FragmentActivity
 import androidx.recyclerview.widget.RecyclerView
+import androidx.viewpager.widget.PagerAdapter
+import androidx.viewpager.widget.ViewPager
 import com.checkin.app.checkin.Home.model.NearbyRestaurantModel
 import com.checkin.app.checkin.R
 import com.checkin.app.checkin.Utility.BaseViewHolder
+import com.rd.PageIndicatorView
+import com.rd.animation.type.AnimationType
+import java.net.URL
+import java.util.*
+import android.graphics.BitmapFactory
+import android.graphics.Bitmap
+import android.nfc.Tag
+import android.widget.TextView
+import butterknife.BindView
+import com.bumptech.glide.Glide
+
+
+class NearbyRestaurantAdapter(var context: Context) : RecyclerView.Adapter<BaseViewHolder<Any>>() {
+
+
 
-class NearbyRestaurantAdapter : RecyclerView.Adapter<BaseViewHolder<Any>>() {
     var data: List<NearbyRestaurantModel> = emptyList()
 
     fun updateData(newData: List<NearbyRestaurantModel>) {
+
         data = newData
         notifyDataSetChanged()
     }
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder<Any> {
         val view = LayoutInflater.from(parent.context).inflate(viewType, parent, false)
-        return if (viewType == R.layout.item_home_restaurant_banner) NearbyRestaurantViewHolder(view) as BaseViewHolder<Any> else AdvertisementViewHolder(view)
+        return if (viewType == R.layout.item_home_restaurant_banner) NearbyRestaurantViewHolder(view)
+                as BaseViewHolder<Any> else AdvertisementViewHolder(view)
     }
 
-    override fun getItemCount(): Int = data.count() + 1
+    override fun getItemCount(): Int = data.count()
```

Comment:
Logic is wrong here. Total count of item in RV is `count(nearbyRestaurants) + count(adBanner)`